cmake/check_compiler_flag.cmake

```diff
@@ -54,3 +54,43 @@ FUNCTION(MY_CHECK_AND_SET_COMPILER_FLAG flag_to_set)
     ENDIF()
   ENDFOREACH()
 ENDFUNCTION()
+
+FUNCTION(MY_CHECK_AND_SET_COMPILER_FLAG_CC flag_to_set)
```

Comment:
This needs to be MY_CHECK_AND_SET_COMPILER_FLAG_C , not MY_CHECK_AND_SET_COMPILER_FLAG_CC (not C vs CC at the end) . CMake only has C or CXX for C and C++ , there is no CC, which can be very easily mistaken for having something to do with C++.

Reply:
The name is a nod to the CC variable, which specifies the C compiler. I did this so you can easily search for MY_CHECK_AND_SET_COMPILER_FLAG_CC without getting matches for MY_CHECK_AND_SET_COMPILER_FLAG_CXX. Please let me know if you still want this to be changed.

Reply:
Thanks I understand the nod, and yes, I still want to be changed. If it called WHATEVER_COMPILER_FLAGS_C  literally *everywhere*  in everything CMake related. The only thing CMake does with CC is reading the env.variable, which I'm pretty sure this comes from compatibility with autotools.